### PR TITLE
Added Netlify optimization

### PR DESCRIPTION
**Pull request by SDBrett**
Updated netlify.toml to enabling post processing tasks to improve website optimizations.

Signed-off-by: Brett Johnson <brett@sdbrett.com>

### CODE REVIEW

**Comment by SDBrett** [comment]
In general the settings are changes to enable post processing and improve overall website performance. These settings overwrite post processing settings configured through the Netlify UI.

[build.processing]
  skip_processing = false -> Enable processing
[build.processing.css]
  bundle = true -> Concatenate consecutive CSS files together to reduce HTTP requests.
  minify = true -> Run CSS through a minifier to reduce file size.
[build.processing.js]
  bundle = true -> Concatenate consecutive JS files together to reduce HTTP requests.
  minify = true -> Run JS through a minifier to reduce file size.
[build.processing.images]
  compress = true -> Run all images through lossless image compression.

### Netlify documentation

Optimization blog post: https://www.netlify.com/blog/2019/08/05/control-your-asset-optimization-settings-from-netlify.toml/
How to guide: https://docs.netlify.com/configure-builds/file-based-configuration/#post-processing

**Comment by jpeach** [comment]
@SDBrett Those comments are great, let's pull them into `netlify.toml`.

File: netlify.toml [hunk 1]
```diff
@@ -2,3 +2,17 @@
   base = "site/"
   command = "jekyll build"
   publish = "site/_site"
+[build.processing]
+  skip_processing = false # -> Enable processing
+[build.processing.css]
+  bundle = true # -> Concatenate consecutive CSS files together to reduce HTTP requests.
```
Comment (davecheney) [inline netlify.toml]: Jekyll already runs the css through sass and produces a single style css. Is this needed?

File: netlify.toml [hunk 2]
```diff
@@ -2,3 +2,17 @@
...
+  minify = true # -> Run CSS through a minifier to reduce file size.
+[build.processing.js]
+  bundle = true # -> Concatenate consecutive JS files together to reduce HTTP requests.
```
Comment (davecheney) [inline netlify.toml]: Same as above

Comment (SDBrett) [reply netlify.toml]: There are a few differences noticed on the [test site](https://deploy-preview-1815--projectcontour.netlify.com/guides/deploy-aws-nlb/)  that Netlify build for this PR which indicate that the additional post processing has an effect.

A basic navigation test across 5 pages on the site with caching enabled
Live site: 104 requests, 3.45MB transferred
Test site: 85 requests, 3.34MB transferred

The match height JS is not indemnified on the live site but is on the test site. Which is part of the goal, provide a second level of optimization without needing to add plugins to Jekyll or create and manage a pipeline to perform post build processing.

There is a limitation of how much if the JS can be minified as a number of scripts are pulled from external sources on page load and not part of the processing.

File: netlify.toml (hunk 1 above)
Comment (SDBrett) [reply netlify.toml]: After reviewing the live site and test site from Netlify I can see that the CSS bundling and minify process from Netlify doesn't appear to have any impact.

I can remove those from the toml file if you would like

**Comment by SDBrett** [comment]
Closing PR to re-evaluate approach

**Comment by davecheney** [comment]
@SDBrett 1.0 is out, please feel free to send a replacement PR.

**Comment by davecheney** [comment]
@SDBrett 1.0 is out, please feel free to send a replacement PR.

**Comment by SDBrett** [comment]
Will send replacement PR in the next day or so

**Comment by davecheney** [comment]
Gentle ping

**Pull request closed without merging**